.github/workflows/tests.yml

```diff
@@ -27,13 +27,49 @@ jobs:
       - name: Rebase to current master
         run: git rebase origin/master
 
+      - name: Prepare
+        run: |
+          # Put the /secrets in place which we'll mount to the tasks-container
+          mkdir -p ~/secrets
+          echo '${{ secrets.COCKPITUOUS_TOKEN }}' > ~/.config/github-token
+          echo '${{ secrets.SSH_KNOWN_HOSTS }}' > ~/secrets/known_hosts
+          echo '${{ secrets.SINK_SSH_CONFIG }}' > ~/secrets/ssh-config
+          echo '${{ secrets.SINK_ID_RSA }}' > ~/secrets/id_rsa
+          echo '${{ secrets.SINK_ID_RSA_PUB }}' > ~/secrets/id_rsa.pub
+          chmod 400 ~/secrets/*
+          # change permissions to user in cockpit-tasks container
+          sudo chown 1111:1111 ~/secrets/*
+
       - name: Build and test tasks container if it changed
         run: |
           changes=$(git diff --name-only origin/master..HEAD -- tasks/)
           if [ -n "${changes}" ]; then
               # Run podman as root, as podman is missing slirp4netns by default, and does not have overlayfs by default
               sudo make tasks-container
-              # This does not do much yet, just iterate to the AMQP queue 30 times and exit
-              # TODO: Inject a workload that does not need /dev/kvm
-              sudo tasks/run-local.sh
+
+              # Start the amqp/tasks pod in the background and supply working task secrets
+              sudo tasks/run-local.sh -d -s ~/secrets -t ~/.config/github-token
+
+              # trigger a test
+              PRN=$(echo "$GITHUB_REF" | cut -f3 -d '/')
+              # wait until bots checkout
+              sudo podman exec -i cockpituous-tasks sh -exc "
+              until [ -f cockpit-project/bots/tests-trigger ]; do
+                  sleep 5;
+              done;
+              cd cockpit-project/bots;
+              ./tests-trigger -f --repo cockpit-project/cockpituous $PRN unit-tests;
+              until (./tests-scan --repo cockpit-project/cockpituous -p $PRN -dv 2>/dev/null | grep -q 'unit-tests'); do
+                 ./tests-scan --repo cockpit-project/cockpituous -p $PRN -dv;
```

Comment:
This seems unnecessary? The `until` condition already calls the exact same command. Also, let's introduce a timeout. Maybe something like
```sh
for retry in `seq 10`; do
    OUT=$(tests-scan..)
    [ "${OUT%unit-tests*}" = "$OUT" ] || break
    [... message, sleep.. ]
done
```

Also, nitpick: Not necessary to have the `;` line endings, as these are already in separate lines.

Reply:
mostly for debugging, i'll remove it. agreed on the timeout